h2o-core/src/main/java/water/persist/PersistHTTP.java

```diff
@@ -0,0 +1,145 @@
+package water.persist;
+
+import com.google.common.io.ByteStreams;
+import org.apache.http.Header;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpHead;
+import org.apache.http.impl.client.HttpClientBuilder;
+import water.H2O;
+import water.Key;
+import water.MemoryManager;
+import water.Value;
+import water.fvec.FileVec;
+import water.fvec.HTTPFileVec;
+import water.fvec.Vec;
+import water.util.FrameUtils;
+import water.util.HttpResponseStatus;
+import water.util.Log;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Implementation of the Persist interface for HTTP/HTTPS data sources
+ * Only subset of the API is supported.
+ */
+public class PersistHTTP extends Persist {
+
+  @Override
+  public byte[] load(Value v) throws IOException {
+    final byte[] b = MemoryManager.malloc1(v._max);
+    final Key k = v._key;
+    final long offset = (k._kb[0] == Key.CHK) ? FileVec.chunkOffset(k) : 0L;
+
+    URI source = decodeKey(k);
+    HttpGet req = new HttpGet(source);
+    req.setHeader(HttpHeaders.RANGE, "bytes=" + offset + "-" + (offset+v._max-1));
+
+    HttpClient client = HttpClientBuilder.create().build();
```

Comment:
Building the client each and every time might be considered expensive. Yet this is no bottleneck for H2O and I'd leave it here, instead of having it in the memory instantiated for the whole lifetime of PersistHTTP instance.

Reply:
Good point @Pscheidl! I also noticed I am not closing the client and the response :)